src/app/blog/(posts)/godot-choosing-network-protocol/page.mdx

```diff
@@ -0,0 +1,92 @@
+  can easily change it later.
+</Info>
+
+## Choosing the Right Networking Protocol
+
+When setting up multiplayer in Godot, you need to choose the appropriate networking protocol by choosing a [MultiplayerPeer](https://docs.godotengine.org/en/stable/classes/class_multiplayerpeer.html#class-multiplayerpeer) implementation based on your game's requirements. Godot provides several options, each with its own characteristics and use cases:
+
+### `ENetMultiplayerPeer` (ENet)
+
+- Ideal for games that require fast-paced, real-time multiplayer gameplay.
+- Uses the [ENet library](http://enet.bespin.org/), which is designed for reliable, low-latency communication over UDP.
+- Supports both client-server and peer-to-peer (mesh) architectures.
+- Provides multiple channels for prioritizing different types of data.
+- Suitable for desktop and mobile platforms.
+- Pros:
+  - Low latency and reliable communication.
+  - Efficient bandwidth usage.
+  - Supports multiple channels for data prioritization.
+- Cons:
+  - Does not work with web.
+  - May be blocked on many firewalls because of UDP.
+  - Requires extra steps to [configure DTLS encryption](https://godotengine.org/article/enet-dtls-encryption/) for secure transport.
+
+For more information, see the [ENetMultiplayerPeer documentation](https://docs.godotengine.org/en/stable/classes/class_enetmultiplayerpeer.html).
+
+### `WebSocketMultiplayerPeer` (WebSocket)
+
+- Ideal for games that need to support web browsers and doesn't require peer-to-peer.
+- Uses the [WebSocket protocol](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API), which is well standardized and works with most proxies & firewalls.
+- Supports secure communication using TLS/SSL.
+- Pros:
+  - Wide browser support and compatibility.
+  - Supports secure communication with TLS/SSL.
+  - Simple protocol.
+- Cons:
+  - Higher latency compared to UDP-based protocols like ENet and WebRTC.
+  - Does not support peer-to-peer.
+
+For more information, see the [WebSocketMultiplayerPeer documentation](https://docs.godotengine.org/en/stable/classes/class_websocketmultiplayerpeer.html).
+
+### `WebRTCMultiplayerPeer` (WebRTC)
+
+- Ideal for games that require peer-to-peer communication and support for web browsers.
+- Uses the [WebRTC protocol](https://webrtc.org/), which enables direct peer-to-peer connections.
+- Supports NAT traversal in order to allow peer-to-peer connections from behind a firewall.
+- Suitable for web-based games or games that require advanced peer-to-peer functionality.
+- Pros:
+  - Enables direct peer-to-peer communication.
+- Cons:
+  - More complex to set up and manage compared to other protocols.
+  - May have compatibility issues with certain platforms or network configurations.
+
+For more information, see the [WebRTCMultiplayerPeer documentation](https://docs.godotengine.org/en/stable/classes/class_webrtcmultiplayerpeer.html).
+
+## Conclusion
```

Comment:
An idea, it might be good to add some examples

```
- **Competitive first person shooter for Windows**: Using ENet is likely the best option here. ENet checks the boxes  for the need of real-time gameplay, and server-authoritative physics to prevent some types of client cheats. If the game was cooperative without worry of cheating, WebRTC could also be a good option.
- **Card game in the browser**: In this case, WebSockets could be a good solution. Since latency isn't the primary focus in the turn-based genre, better browser support for the WebSocket spec will come in quite helpful.
```